test/services/TwoStepVerificationServiceSpec.scala

```diff
@@ -198,17 +205,26 @@ class TwoStepVerificationServiceSpec extends UnitSpec with MockitoSugar with Wit
         implicit val authContext = AuthContext(loggedInUser, principal, None)
 
         when(ruleContext.currentCoAFEAuthority).thenReturn(Future.successful(CoAFEAuthority(None, "", "")))
-        when(ruleContext.activeEnrolments).thenReturn(Future.successful(enrolments))
+        when(ruleContext.activeEnrolmentKeys).thenReturn(Future.successful(enrolments))
         when(ruleContext.isAdmin).thenReturn(Future.successful(isAdmin))
+        val activeGGEnrolments = enrolments.map(GovernmentGatewayEnrolment(_, Seq(), "Activated")).toSeq
+        when(ruleContext.activeEnrolments).thenReturn(Future.successful(activeGGEnrolments))
+        val credentialRole = if (isAdmin) "User" else "Assistant"
+
+        when(ruleContext.userDetails).thenReturn(Future.successful(UserDetails(Some(CredentialRole(credentialRole)), "affinityGroup")))
+
         when(ruleContext.enrolments).thenReturn(Future.successful(Seq.empty[GovernmentGatewayEnrolment]))
         when(twoStepVerificationThrottleMock.isRegistrationMandatory(expectedRuleName, userId)).thenReturn(Future.successful(isMandatory))
+        when(auditConnectorMock.sendEvent(any[AuditEvent])(any[HeaderCarrier], any[ExecutionContext])).thenReturn(Future.successful(mock[AuditResult]))
 
         val result = await(twoStepVerification.getDestinationVia2SV(BusinessTaxAccount, ruleContext, auditContext))
 
         result shouldBe Some(expectedRedirectLocation())
         verify(ruleContext, atLeastOnce()).currentCoAFEAuthority
-        verify(ruleContext, Mockito.atMost(2)).activeEnrolments
+        verify(ruleContext, Mockito.atMost(2)).activeEnrolmentKeys
```

Comment:
these calls are made either once or twice, depending on the scenario defined in the table above